server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -958,6 +1025,28 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
             if (format != null) {
                 throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] doesn't support formats.");
             }
+            if (this.script != null) {
+                // values don't live in source - either pull from dv, or re-calculate
+                if (hasDocValues()) {
+                    return new DocValueFetcher(DocValueFormat.RAW, context.getForField(this));
+                }
+                return new ValueFetcher() {
+                    LeafReaderContext ctx;
+
+                    @Override
+                    public void setNextReader(LeafReaderContext context) {
+                        this.ctx = context;
+                    }
+
+                    @Override
+                    public List<Object> fetchValues(SourceLookup lookup) {
+                        List<Object> values = new ArrayList<>();
+                        script.executeAndIndex(context.lookup(), ctx, lookup.docId(), values::add);
+                        values.sort(Comparator.comparingLong(a -> (Long) a));
```

Comment:
I believe we're ok with getting the values back in whatever order. We don't, for example, sort the values from source in SearchValueFetcher.